internal/services/cdn/cdn_frontdoor_rule_resource.go

```diff
@@ -726,7 +726,7 @@ func resourceCdnFrontDoorRuleUpdate(d *pluginsdk.ResourceData, meta interface{})
 	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.FrontDoorRuleID(d.Id())
+	id, err := parse.FrontDoorRuleIDInsensitively(d.Id())
```

Comment:
this is coming from the Resource ID so we can parse this case-sensitively:

```suggestion
	id, err := parse.FrontDoorRuleID(d.Id())
```

Reply:
Fixed.